x-pack/legacy/plugins/reporting/server/lib/get_user.ts

```diff
@@ -4,14 +4,17 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export function getUserFactory(server) {
-  return async request => {
+import { Request } from 'hapi';
+import { RequestFacade, ServerFacade } from '../../types';
+
+export function getUserFactory(server: ServerFacade) {
+  return async (request: RequestFacade) => {
     if (!server.plugins.security) {
       return null;
     }
 
     try {
-      return await server.plugins.security.getUser(request);
+      return await server.plugins.security.getUser((request as unknown) as Request);
```

Comment:
Our `RequestFacade` request has to be forced into a `Hapi.Request` so it can be passed along to this function from the external security plugin.

Reply:
but security plugin expects `Hapi.Request`. Potentially it can throw accessing any property not covered by `RequestFacade`. Probably it's safer for `RequestFacade` to enhance `Hapi.Request` types?
```ts
export type RequestFacade = Legacy.Request & {
  payload: JobParamPostPayload | GenerateExportTypePayload;
  query: ReportingRequestQuery;
  pre: ReportingRequestPre;
};
// make_request_facade.ts 
export function makeRequestFacade(request: Legacy.Request) {
  return request as RequestFacade;
}
```

Reply:
The goal is to understand which properties of `req` we really depend on, and have an interface that accurately describes the dependencies. By using that method, we are not getting closer to that goal, so I don't think I'd want to do that.

Reply:
Correctly typed code solves the same problem, but YMMV. 
If you stick to the current solution, I'd suggest to change the code not to affect 3rd party code:
```ts
type RequestFacade {
  getRawRequest: () => Legacy.Request;
  ...
}
return await server.plugins.security.getUser(request.getRawRequest());
```

Reply:
resolved in 80e0fcc

Reply:
Oops, I mean cb40fa3896

See https://github.com/elastic/kibana/pull/54082#discussion_r366589274